### PR TITLE
Log Results Sorting

### PR DESCRIPTION
**Pull request by brandon-pereira**
Adds support for ordering logs by clicking on the respective header. 

- Toggles between three states (unset, asc, desc).
- Leverages the column aliases correctly so if aliases have special characters (ex `DURATION as "Duration (ms)"`) they are correctly ordered as expected
- Supports nested tables 
- Supports query params (on search page)
- Adds tests for db row table and sorting logic

Demo:

https://github.com/user-attachments/assets/46aa0299-1ed3-480b-a35f-315d921e47ba

Fixes HDX-2317

### CODE REVIEW

**Comment by pulpdrew** [comment]
One edge case to be aware of is that using JSON schema, this will break if the user doesn't specify a type for the property in the select query:

Screenshot 2025-10-01 at 4 47 32 PM

The error message is pretty reasonable so this might not be the biggest deal, but if it is possible to automatically solve or prevent the issue when they click the column header, that could be a nice improvement.

**Comment by brandon-pereira** [comment]
@pulpdrew I disabled sorting if its a JSON Dynamic type (for now) - I will make a ticket to improve this in the future.

@MikeShi42 I added support to the other tables!

File: packages/app/src/ClickhousePage.tsx [hunk 1]
```diff
@@ -363,6 +363,7 @@ function InsertsTab({
           </Text>
           <DBTableChart
             config={{
+              dateRange: searchedTimeRange,
```
Comment (brandon-pereira) [inline packages/app/src/ClickhousePage.tsx:366]: This fixes the clickhouse inserts table, its currently broken in `main`

Comment (brandon-pereira) [reply packages/app/src/ClickhousePage.tsx:366]: @pulpdrew

Comment (pulpdrew) [reply packages/app/src/ClickhousePage.tsx:366]: Ah, thanks for the heads up! Assigned the issue to you 👍

File: packages/app/src/components/DBTable/TableHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import cx from 'classnames';
+import { Button, Group, Text, UnstyledButton } from '@mantine/core';
+import {
+  IconChevronDown,
+  IconChevronUp,
```
Comment (elizabetdev) [inline packages/app/src/components/DBTable/TableHeader.tsx]: Since we’re already using chevron icons to represent toggleable elements, should we use `IconArrowDown` and `IconArrowUp` instead?

Also, maybe show the arrow on the current sort state when the page loads?

Comment (brandon-pereira) [reply packages/app/src/components/DBTable/TableHeader.tsx]: Good idea, I switched the icons!

I also improved the initial load logic, it should work more reliably now!

**Review by dhable — APPROVED** [review]
LGTM
